src/com/google/common/css/MultipleMappingSubstitutionMap.java

```diff
@@ -76,5 +77,5 @@ public static ValueWithMappings createForSingleMapping(String key, String value)
    * associated with this substitution. These mappings are available as
    * {@link ValueWithMappings#mappings}.
    */
-  public ValueWithMappings getValueWithMappings(String key);
+  public ValueWithMappings getValueWithMappings(String key, @Nullable Map<String, String> inputMappings);
```

Comment:
I think that this make sense.

And to block the generation of of random ID that exists in input map, Is it ok for you to use an `outputValueBlacklist` in `SubstitutionMap` class (see change `RenamingType.java`) ?